src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -158,6 +151,26 @@ private async Task<Dictionary<ImmutableArray<byte>, DocumentId>> ComputeContentH
         return result;
     }
 
+    private AsyncLazy<ProjectStateChecksums> LazyChecksums
+    {
+        get
+        {
+            _lazyChecksums ??= AsyncLazy.Create(static (self, cancellationToken) => self.ComputeChecksumsAsync(cancellationToken), arg: this);
```

Comment:
i'm oogy on this.  there's a race here where multipel threads could see null and assign a new value.  I'm not 100% convinced that computing the checksum is guaranteed to produce the exact same result both times.

can you make this an interlocked CAS so only one wins?

Reply:
Sure, I can do that. I was definitely under the impression that computing the checksum would generate the same values if it happened to get unlucky and run multiple times.

Reply:
i want to believe that that would be the case... but i don't think it's proven to be true.

Reply:
OK, switched to use Interlocked.CompareExchange